lib/ansible/modules/network/nxos/nxos_banner.py

```diff
@@ -111,7 +111,19 @@ def map_obj_to_commands(want, have, module):
 
 
 def map_config_to_obj(module):
-    output = run_commands(module, ['show banner %s' % module.params['banner']], False)[0]
+    command = 'show banner %s' % module.params['banner']
+    output = 'json'
+    cmds = [{
+        'command': command,
+        'output': output,
+    }]
+    output = run_commands(module, cmds, False)
+    if len(output) == 0 or len(output[0]) == 0:
+        # If we get here the platform does not
+        # support structured output.  Resend as
+        # text.
+        cmds[0]['output'] = 'text'
+        output = run_commands(module, cmds, False)
```

Comment:
@mikewiebe What about we define this in a function `execute_show_command(command, module)`?

Reply:
Ack.  Good suggestion